### PR TITLE
Add select all file when upload file to remote

### PR DESCRIPTION
**Pull request by xxNull-lsk**

### CODE REVIEW

**Comment by przybylski** [comment]
Instead of checkbox IMO this should be an option in actionbar.
Also please rebase your changes

**Comment by tobiasKaminsky** [comment]
A screenshot of the change would be very nice.
Especially @jancborchardt does not compile by himself nor can really "read" sourcecode (if I am not wrong [of course no affront]).

**Comment by tobiasKaminsky** [comment]
The translation is done via transifex so it is not neccessary to do the translation here (in fact it will get overridden). So, it is enough to have the english values.

**Comment by xxNull-lsk** [comment]
I will rebase.screenshot is here.

**Comment by xxNull-lsk** [comment]
Already design.Please recheck.

**Comment by jancborchardt** [comment]
Since for other languages the text will be considerably longer, it’s better to use the »Select all« icon here: https://materialdesignicons.com/icon/select-all

**Comment by xxNull-lsk** [comment]
Chenged.

**Comment by xxNull-lsk** [comment]
Already changed "select all" from text to icon.

**Comment by jancborchardt** [comment]
Cool! Btw, any reason the checkboxes are not material design style? Is that coming in a separate PR @tobiasKaminsky @AndyScherzinger?

**Comment by tobiasKaminsky** [comment]
@jancborchardt I think this is all handled by one of @AndyScherzinger branches.
As this PR is forking from master it is not there already.
But it should (and will) be there once the material branches are merged.

**Comment by AndyScherzinger** [comment]
@jancborchardt yes that is still laying around waiting to be merged #1209 so pleeeease let's get this merged. Getting asked about the checkboxes and primary actions every second day in another PR is kind of annoying :wink: 

@xxNull-lsk can you check the icon again, this should be put into the different folder for icons (mdpi, hdpi, xhdpi and xxhdpi) in their respective sizes (can usually be downloaded in these sizes. This is necessary to have sharp and crisp icons on the different solutions. Else Android scales the icon and it will be blurred (an seen in your screenshot). Besides that: really looking good :) :+1:

**Comment by jancborchardt** [comment]
@AndyScherzinger hehe, sorry ;D unfortunately I can not review / QA this, it’s what @davivel @masensio @rperezb need to do, at least as a final pass.

File: res/values/strings.xml [hunk 1]
```diff
@@ -394,5 +394,6 @@
 
     <string name="share_sharee_unavailable">Sorry, your server version does not allow share with users within clients.
         \nPlease contact your administrator</string>
+    <string name="file_list_select_all">Select All</string>
```
Comment (przybylski) [inline res/values/strings.xml]: there is a better place to put this, around 74th line

File: src/com/owncloud/android/ui/fragment/LocalFileListFragment.java [hunk 2]
```diff
@@ -98,6 +98,18 @@ public void onActivityCreated(Bundle savedInstanceState) {
         
         Log_OC.i(TAG, "onActivityCreated() stop");
     }
+
+    public void CheckAllFile(boolean value)
```
Comment (przybylski) [inline src/com/owncloud/android/ui/fragment/LocalFileListFragment.java]: checkAllFiles <- lowercase at first char & plural

check is more of UI style naming, maybe selectAllFiles

File: src/com/owncloud/android/ui/fragment/LocalFileListFragment.java [hunk 3]
```diff
@@ -98,6 +98,18 @@ public void onActivityCreated(Bundle savedInstanceState) {
...
+    {
+        ListView lst = (ListView)getListView();
+        int pos;
+        for (pos = 0; pos < lst.getCount(); pos++) {
```
Comment (przybylski) [inline src/com/owncloud/android/ui/fragment/LocalFileListFragment.java]: there is no need for `pos` to be visible outside for loop

File: res/menu/upload_files.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+      xmlns:app="http://schemas.android.com/apk/res-auto">
```
Comment (przybylski) [inline res/menu/upload_files.xml]: Is this right?

Comment (xxNull-lsk) [inline res/menu/upload_files.xml]: Sorry, The lack of copyright information?I don't known how to fix this,

Comment (przybylski) [inline res/menu/upload_files.xml]: i mean app namespace isn't showAsAction part of android namespace ?

File: src/com/owncloud/android/ui/fragment/LocalFileListFragment.java [hunk 5]
```diff
@@ -98,6 +98,17 @@ public void onActivityCreated(Bundle savedInstanceState) {
         
         Log_OC.i(TAG, "onActivityCreated() stop");
     }
+
+    public void SelectAllFiles(boolean value)
```
Comment (przybylski) [inline src/com/owncloud/android/ui/fragment/LocalFileListFragment.java]: Please use camelCase

File: res/menu/upload_files.xml (hunk 4 above)
Comment (xxNull-lsk) [inline res/menu/upload_files.xml]: Same in main_menu.xml.
Anyway already changed in upload_files.xml



Allan@dingyantec.com
 
From: Bartosz Przybylski
Date: 2015-12-18 10:06
To: owncloud/android
CC: Allan
Subject: Re: [android] Add select all file when upload file to remote (#1342)
In res/menu/upload_files.xml:
> @@ -0,0 +1,11 @@
> +<?xml version="1.0" encoding="utf-8"?>
> +<menu xmlns:android="http://schemas.android.com/apk/res/android"
> +      xmlns:app="http://schemas.android.com/apk/res-auto">

i mean app namespace isn't showAsAction part of android namespace ?

Comment (AndyScherzinger) [inline res/menu/upload_files.xml]: This should be "app"! See for example here: http://stackoverflow.com/questions/26377305/item-with-appshowasaction-not-showing This is simply due to the fact that we have implemented material design and its action bar via the AppCompat lib not the "native" action bar which wouldn't be material all the way back to version 14. Thus it need to be the re-auto namespace (!)

Comment (przybylski) [inline res/menu/upload_files.xml]: Yeah that what I was thinking

Comment (AndyScherzinger) [inline res/menu/upload_files.xml]: @xxNull-lsk Can you then bring back app:... ? :smiley:

Comment (xxNull-lsk) [inline res/menu/upload_files.xml]: @Andy Scherzinger ok



Allan@dingyantec.com
 
From: Andy Scherzinger
Date: 2015-12-18 17:01
To: owncloud/android
CC: Allan
Subject: Re: [android] Add select all file when upload file to remote (#1342)
In res/menu/upload_files.xml:
> @@ -0,0 +1,11 @@
> +<?xml version="1.0" encoding="utf-8"?>
> +<menu xmlns:android="http://schemas.android.com/apk/res/android"
> +      xmlns:app="http://schemas.android.com/apk/res-auto">

@xxNull-lsk Can you then bring back app:... ?

**Comment by dankeder** [comment]
Hi all,
I'm new here, so please bear with me, I'm not an Android dev :). I stumbled upon this PR when searching for how to upload all files in a folder. It'd be awesome if this feature was merged, it would be super useful to many people.

I took the liberty of rebasing the branch of this PR onto the current master + squashing the five commits into one (they were just fixing stuff made in the first one). The rebased branch can be found here if anyone is interested:

https://github.com/dankeder/owncloud-android/commits/select-all-files

I can create a new PR from the rebased branch if it will help to get this merged.

**Comment by Basskrapfen** [comment]
Is this change still alive? Would be really happy to see this feature in a future release.

**Comment by pixolex** [comment]
Please, put the select all button in the app.

**Comment by davigonz** [comment]
> Please, put the select all button in the app.

We are currently working on Select all button and should be included for the next release 2.9 but not via this PR since is pretty old, follow up https://github.com/owncloud/android/issues/1727 and https://github.com/owncloud/android/pull/2242

**Comment by jonferreira** [comment]
I'm a bit lost with all that's happening but was this implemented or not?

I just installed owncloud on my S8 and want to upload all the existing photos but there doesn't appear to be a Select All button

**Comment by pvint** [comment]
I just installed Owncloud and tried to back up things from my phone, but without folder upload or "Select All" it's simply too tedious to upload thousands of files.

Back to NextCloud :(

**Comment by davigonz** [comment]
This PR is too old, we will include this feature via https://github.com/owncloud/android/pull/2404

**Pull request closed without merging**